SQL/admin_import_2018-02-03.py

```diff
@@ -0,0 +1,81 @@
+#python admin_import_2018-02-03.py "localhost" "root" "password" "feedback" "SS13_admin" "SS13_admin_ranks"
+#
+#This script performs no error-correction, improper configurations of admins.txt or admin_ranks.txt will cause either breaking exceptions or invalid table rows
+#It's safe to run this script with your game server(s) active.
+
+
+import MySQLdb
+import argparse
+import re
+import sys
+
+def parse_text_flags(text, previous):
+    flag_values = {"BUILDMODE":1, "BUILD":1, "ADMIN":2, "REJUVINATE":2, "REJUV":2, "BAN":4, "FUN":8, "SERVER":16, "DEBUG":32, "POSSESS":64, "PERMISSIONS":128, "RIGHTS":128, "STEALTH":256, "POLL":512, "VAREDIT":1024, "SOUNDS":2048, "SOUND":2048, "SPAWN":4096, "CREATE":4096, "AUTOLOGIN":8192, "AUTOADMIN":8192, "DBRANKS":16384}
+    flags_int = 8192
+    exclude_flags_int = 0
+    flags = text.split(" ")
+    if flags:
+        for flag in flags:
+            if flag[1:] in ("@", "prev"):
+                flags_int = previous
+                continue
+            sign = flag[:1]
+            if flag[1:] in ("EVERYTHING", "HOST", "ALL"):
+                if sign is "+":
+                    flags_int = 65535
+                else:
+                    exclude_flags_int = 65535
+                continue
+            if flag[1:] in flag_values:
+                if sign is "+":
+                    flags_int += flag_values[flag[1:]]
+                else:
+                    exclude_flags_int += flag_values[flag[1:]]
+    flags_int = max(min(65535, flags_int), 0)
+    exclude_flags_int = max(min(65535, exclude_flags_int), 0)
+    return flags_int, exclude_flags_int
+
+if sys.version_info[0] < 3:
+    raise Exception("Python must be at least version 3 for this script.")
+parser = argparse.ArgumentParser()
+parser.add_argument("address", help="MySQL server address (use localhost for the current computer)")
+parser.add_argument("username", help="MySQL login username")
+parser.add_argument("password", help="MySQL login username")
+parser.add_argument("database", help="Database name")
+parser.add_argument("admintable", help="Name of the current admin table (remember prefixes if you use them)")
+parser.add_argument("rankstable", help="Name of the current admin ranks (remember prefixes)")
+args = parser.parse_args()
+db=MySQLdb.connect(host=args.address, user=args.username, passwd=args.password, db=args.database)
+cursor=db.cursor()
+ranks_table = args.rankstable
+admin_table = args.admintable
+with open("..\\config\\admin_ranks.txt") as rank_file:
+    previous = 0
+    for line in rank_file:
+        if line.strip():
+            if line.startswith("#"):
+                continue
+            matches = re.match("(.+)\\b\\s+=\\s*(.*)", line)
+            flags = parse_text_flags(matches.group(2), previous)
+            previous = flags[0]
+            cursor.execute("INSERT INTO {0} (rank, flags, exclude_flags, can_edit_flags) VALUES ('{1}', {2}, {3}, 0)".format(ranks_table, matches.group(1), flags[0], flags[1]))
```

Comment:
![image](https://user-images.githubusercontent.com/7069733/36628859-2dfda696-1903-11e8-881f-8b9ffc819f4e.png)

This is with the default repo admins and ranks.

I don't think this is correct.